components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineView.kt

```diff
@@ -639,20 +643,45 @@ class SystemEngineView @JvmOverloads constructor(
 
     override fun canScrollVerticallyDown() = session?.webView?.canScrollVertically(1) ?: false
 
-    @Suppress("Deprecation")
-    // TODO remove suppression when fixed: https://github.com/mozilla-mobile/android-components/issues/888
     override fun captureThumbnail(onFinish: (Bitmap?) -> Unit) {
         val webView = session?.webView
+        if (webView == null) {
+            onFinish(null)
+            return
+        }
 
-        val thumbnail = if (webView == null) {
-            null
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
+            createThumbnailUsingDrawingView(webView, onFinish)
         } else {
-            webView.buildDrawingCache()
-            val outBitmap = webView.drawingCache?.let { cache -> Bitmap.createBitmap(cache) }
-            webView.destroyDrawingCache()
-            outBitmap
+            createThumbnailUsingPixelCopy(webView, onFinish)
         }
-        onFinish(thumbnail)
+    }
+
+    private fun createThumbnailUsingDrawingView(view: View, onFinish: (Bitmap?) -> Unit) {
+        val outBitmap = Bitmap.createBitmap(view.width, view.height, Bitmap.Config.ARGB_8888)
+        val canvas = Canvas(outBitmap)
+        view.draw(canvas)
+        onFinish(outBitmap)
+    }
+
+    @TargetApi(Build.VERSION_CODES.O)
+    private fun createThumbnailUsingPixelCopy(view: View, onFinish: (Bitmap?) -> Unit) {
+        val out = Bitmap.createBitmap(view.width, view.height, Bitmap.Config.ARGB_8888)
+        val viewRect = getRectWithViewLocation(view)
+        val window = (context as Activity).window
+
+        PixelCopy.request(window, viewRect, out, { copyResult ->
+            val result = if (copyResult == PixelCopy.SUCCESS) out else null
+            onFinish(result)
+        }, handler)
+    }
+
+    private fun getRectWithViewLocation(view: View) = Rect().apply {
```

Comment:
Not a big deal but seems like nice candidate for extension function. :)

Reply:
It's good point, thanks!
Done it!